### PR TITLE
Create a comment on a post and fix profiles pictures.

### PR DESCRIPTION
**Pull request by BlazeDex**
# Description

- [x] The users now can create a comment on any post created.
- [x] When a created user does not have any profile picture, they are assigned a default image.

## Screenshots

Comment
createdcomment

### CODE REVIEW

**Review by MauricioNC — APPROVED** [review]

**Review by juperameza — APPROVED** [review]
Lgtm 
Just try to add cable action to let the user comment in real time 😃😃😃🦜🦜🦜
Also nice test images

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    if @comment.save
+      redirect_to post_path(@post), notice: 'Comment was successfully created'
+    else
+      redirect_to post_path(@post), alert: 'Failed to create comment!'
+    end
+  end
+
+  def reply
+    @post = Post.find(params[:post_id])
```
Comment (JuanequeX) [inline app/controllers/comments_controller.rb]: this is the same instruction that you set in the `find_post` private method

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+class CommentsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :find_post, only: [:create]
```
Comment (JuanequeX) [inline app/controllers/comments_controller.rb]: here, you can take place the `reply` metod

File: app/views/comments/_comment.html.erb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<div class="comment">
+  <p><%= comment.body %></p>
+  <% comment.replies.each do |reply| %>
+    <%= render 'comments/comment', comment: reply %>
+  <% end %>
```
Comment (JuanequeX) [inline app/views/comments/_comment.html.erb]: I think that this render does not make any sense to render over the same partial.

**Review by JuanequeX — CHANGES_REQUESTED** [review]

**Review by JuanequeX — COMMENTED** [review]
It would be amazing to integrate system tests for this feature. This can be helpful in verifying that the user flow for creating a comment is successful.

File: app/views/comments/_comment.html.erb (hunk 3 above)
Comment (BlazeDex) [reply app/views/comments/_comment.html.erb]: You're right. The main idea was to generate the form to reply to a parent comment, but in the end I ended up discarding it, and I forgot to remove it when creating the commit. Thx for the observation.

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (BlazeDex) [reply app/controllers/comments_controller.rb]: Updated!

**Review by JuanequeX — APPROVED** [review]

**Pull request merged**